### PR TITLE
BSI formula add to heartpy.py

### PR DESCRIPTION
**Pull request by IgorWounds**
Added the Baevsky Stress index calculation to measures with the corresponding building blocks of the main formula:

si = AMo / (2 * VR * Mo)
where
R-R interval - the time delay between consecutive heartbeats (assumed to be the inverse value of the bpm)
Mo - mode - most frequent R-R interval value
AMo - mode amplitude - % of the intervals corresponding to Mode
VR - variational range - the difference between min and max R-R intervals

Reference: http://www.cardiometry.net/issues/no10-may-2017/heart-rate-variability-analysis

### CODE REVIEW

**Comment by paulvangentcom** [comment]
Thanks, I'll have a look at it this week!

File: heartpy/heartpy.py [hunk 1]
```diff
@@ -294,6 +295,23 @@ def process(hrdata, sample_rate, windowsize=0.75, report_time=False,
...
+    # Obtain Mode
+    mode_v = mode(working_data['RR_list'])
+    measures['Mo'] = mode_v
+
+    # Obtain VR (variational range)
+    vr = np.max(working_data['RR_list']) - np.min(working_data['RR_list'])
+    measures['VR'] = vr
+
+    # Obtain AMo (mode amplitude %)
+    num_mode = np.count_nonzero(working_data['RR_list'] == mode_v)
```
Comment (robin-glimp) [inline heartpy/heartpy.py:308]: Technically this is correct, but it depends on the type of input data. The RR Intervals have to be bucketed into 50ms buckets according to Kubios and the paper by Baevsky (https://cardiometry.net/media/eng/issues/no10-may-2017/heart-rate-variability-analysis.pdf).

Taking a strict equality sounds wrong to me.

File: heartpy/heartpy.py [hunk 2]
```diff
@@ -294,6 +295,23 @@ def process(hrdata, sample_rate, windowsize=0.75, report_time=False,
                                          desired_sample_rate=high_precision_fs, working_data=working_data)
 
     working_data = calc_rr(working_data['peaklist'], sample_rate, working_data=working_data)
+    
+    # Obtain Mode
+    mode_v = mode(working_data['RR_list'])
```
Comment (robin-glimp) [inline heartpy/heartpy.py:300]: Stupid question, I dont know this library very well yet. but what is the unit of the RR-intervals in  `working_data['RR_list']`. This has to be in seconds for the equation to work properly.

File: heartpy/heartpy.py [hunk 3]
```diff
@@ -294,6 +295,23 @@ def process(hrdata, sample_rate, windowsize=0.75, report_time=False,
...
+    measures['VR'] = vr
+
+    # Obtain AMo (mode amplitude %)
+    num_mode = np.count_nonzero(working_data['RR_list'] == mode_v)
+    amo = (num_mode/len(working_data['RR_list']))*100
+    measures['AMo'] = amo
+
+    # Calculate Baevsky Stress Index
+    si = (amo/(2*vr*mode_v)) * 100
+    measures['SI'] = si
```
Comment (robin-glimp) [inline heartpy/heartpy.py:314]: I would love to see the square root of the SI metric here as well. as this is what Kubios also applies to transform the tailed distribution to a normal distribution (see https://www.kubios.com/hrv-analysis-methods/)

**Review by robin-glimp — CHANGES_REQUESTED** [review]

File: heartpy/heartpy.py (hunk 2 above)
Comment (IgorWounds) [reply heartpy/heartpy.py:300]: I'll need to take a look as it was quite a while from when I issued this request. I forgot if the user can control the unit or if it is hardcoded. Thanks!

**Pull request closed without merging**